### PR TITLE
[wayland] Implement XdgForeign

### PR DESCRIPTION
**Pull request by PolyMeilex**
This also includes a SCTK test client, not sure if we want that in the codebase, but I committed it for now, we can just drop the commit as needed.

### CODE REVIEW

**Review by Drakulix — APPROVED** [review]
LGTM (waiting for a decision regarding the example before merging)

**Comment by PolyMeilex** [comment]
Test client dropped, SCTK sounds like a better place, as there it can serve as both example and a test client at the same time.

**Pull request merged**
